pkg/config/templateComponent.go

```diff
@@ -270,14 +263,17 @@ func (t *TemplateComponent) GenerateConfig(cfgType Type, userdata map[string]any
 					return nil, fmt.Errorf("error %w building collector template for %s",
 						err, t.Kind)
 				}
-				tmpl, err := t.generateCollectorConfig(ct, userdata)
+				tmpl, err := t.generateCollectorConfig(ct, pipeline, userdata)
 				if err != nil {
 					return nil, err
 				}
 				generatedTemplates = append(generatedTemplates, tmpl)
 			case "rules":
 				// a rules template expects the metadata to include environment
 				// information.
+				if pipeline.SignalType != hpsf.CTYPE_HONEY {
```

Comment:
Ohh, is this what fixed the problem with duplicated rules generation?

Reply:
Yes! We were generating rules for every pipeline.